csharp/ql/src/semmle/code/csharp/Preprocessor.qll

```diff
@@ -140,3 +140,72 @@ class NullableDirective extends PreprocessorDirective, @directive_nullable {
 
   override string getAPrimaryQlClass() { result = "NullableDirective" }
 }
+
+/**
+ * A `#line` directive, such as `#line default`, `#line hidden`, or `#line`
+ * directive with line number.
+ */
+class LineDirective extends PreprocessorDirective, @directive_line {
+  /** Gets the succeeding `#line` directive in the file, if any. */
+  LineDirective getSuccLineDirective() {
```

Comment:
Why do we need this?

Reply:
It might come handy if someone is looking for the exact scope of a `#line` directive, which is either the file or until the next `#line` if there's any.